src/frontend/src/tests/lib/services/request-pouh-credential.services.spec.ts

```diff
@@ -0,0 +1,122 @@
+		onSuccess({ Ok: successfulCredentialJWT });
+	}
+}));
+
+describe('request-pouh-credential.services', () => {
+	describe('requestPouhCredential', () => {
+		let addUserCredentialMock: MockInstance;
+		let getUserProfileMock: MockInstance;
+
+		const identity = Ed25519KeyIdentity.generate();
+
+		const initialUserProfile: UserProfile = {
+			version: [],
+			created_timestamp: 1234n,
+			updated_timestamp: 1234n,
+			credentials: []
+		};
+
+		beforeEach(() => {
+			vi.clearAllMocks();
+			vi.resetAllMocks();
+			toastsStore.reset();
+			userProfileStore.reset();
+			userProfileStore.set({ certified: true, profile: initialUserProfile });
+			addUserCredentialMock = vi.spyOn(api, 'addUserCredential');
+			getUserProfileMock = vi.spyOn(api, 'getUserProfile');
+		});
+
+		it('should request credential validate it and store it', async () => {
+			addUserCredentialMock.mockResolvedValueOnce({ Ok: null });
+			const userProfileWithCredential: UserProfile = {
+				version: [],
+				created_timestamp: 1234n,
+				updated_timestamp: 1234n,
+				credentials: [
+					{
+						issuer: 'pouh',
+						verified_date_timestamp: [nowInBigIntNanoSeconds()],
+						credential_type: { ProofOfUniqueness: null }
+					}
+				]
+			};
+			getUserProfileMock.mockResolvedValueOnce({ Ok: userProfileWithCredential });
+
+			const result = await requestPouhCredential({ identity });
+
+			expect(result.success).toBe(true);
+			expect(addUserCredentialMock).toBeCalledTimes(1);
+			expect(addUserCredentialMock).toBeCalledWith({
+				identity,
+				credentialJwt: successfulCredentialJWT,
+				credentialSpec: {
+					credential_type: POUH_CREDENTIAL_TYPE,
+					arguments: []
+				},
+				issuerCanisterId: Principal.fromText(POUH_ISSUER_CANISTER_ID),
+				currentUserVersion: undefined,
+				nullishIdentityErrorMessage: get(i18n).auth.error.no_internet_identity
+			});
+			expect(getUserProfileMock).toBeCalledTimes(1);
+			expect(get(userProfileStore)).toEqual({
+				certified: true,
+				profile: userProfileWithCredential
+			});
+		});
+
+		it('should not store the credential if validating credential fails', async () => {
+			addUserCredentialMock.mockResolvedValueOnce({ Err: { InvalidCredential: null } });
+
+			const result = await requestPouhCredential({ identity });
+
+			expect(result.success).toBe(true);
```

Comment:
success = true is correct here? The backend throws `InvalidCredential`.

Reply:
It's a matter of meaning. In this case, the request succeeded, the validation didn't though. Already a toast is shown with the validation error. But I can change that if you think that makes more sense.

Reply:
Thanks for the explanation!